### PR TITLE
feat:(#97-100) created action menu

### PR DESCRIPTION
**Pull request by ghulamhafeez**
### Implementations

- Created common action menu for table action
- Action menu show on role base
- Unit testing 

### Ui 
image (2)

### Test 
image

### CODE REVIEW

File: src/app/layout/components/ActionsMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+
+import React from 'react';
+import { Button } from "@/components/ui/button"
+import { DotsHorizontalIcon } from '@radix-ui/react-icons';
+import { DropdownMenu, DropdownMenuContent, DropdownMenuItem, DropdownMenuTrigger } from '@radix-ui/react-dropdown-menu';
+
+interface Action {
```
Comment (Monfernape) [inline src/app/layout/components/ActionsMenu.tsx:7]: I think it'd be great if we pass `id` prop as well. Could be used later for identification.

File: src/app/layout/components/ActionsMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        >
+
+          <DotsHorizontalIcon className="w-5 h-5" aria-label='icon' />
+        </Button>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent
+        align="end"
+        className="w-48 bg-white border rounded shadow-lg"
+      >
+        {actions?.map((action, index) => (
```
Comment (Monfernape) [inline src/app/layout/components/ActionsMenu.tsx:33]: Why do need optional chaining here?

File: src/app/member/components/MemberActionRender.test.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { it, describe, expect, vi } from 'vitest'; // Add this import
+import { render, screen } from '@testing-library/react';
+import ActionsMenu from '@/app/layout/components/ActionsMenu';
+import userEvent from "@testing-library/user-event";
```
Comment (Monfernape) [inline src/app/member/components/MemberActionRender.test.tsx:4]: Library imports go before local imports.

File: src/app/member/components/MemberActionRender.test.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+describe('ActionsMenu Component', () => {
+  const actionsMock = [
+    { title: 'Action 1', onClick: vi.fn() },
+    { title: 'Action 2', onClick: vi.fn() },
+  ];
+
+  it('calls the correct action when an item is clicked', async () => {
+    render(<ActionsMenu actions={actionsMock} />);
+
+    await userEvent.click(screen.getByLabelText('icon'));
```
Comment (Monfernape) [inline src/app/member/components/MemberActionRender.test.tsx:15]: How does this work? What's `icon`?

**Review by Monfernape — APPROVED** [review]
Minor comments but looks great otherwise.

File: src/app/member/components/MemberActionRender.test.tsx (hunk 4 above)
Comment (ghulamhafeez) [reply src/app/member/components/MemberActionRender.test.tsx:15]: I needed to get this icon `<DotsHorizontalIcon className="w-5 h-5" aria-label='icon' />`, so I gave it a label and used that to get it.

File: src/app/member/components/MemberActionRender.test.tsx (hunk 3 above)
Comment (ghulamhafeez) [reply src/app/member/components/MemberActionRender.test.tsx:4]: Okay

File: src/app/layout/components/ActionsMenu.tsx (hunk 2 above)
Comment (ghulamhafeez) [reply src/app/layout/components/ActionsMenu.tsx:33]: I needed it at the time, but I don't anymore. I will remove it now.

File: src/app/layout/components/ActionsMenu.tsx (hunk 1 above)
Comment (ghulamhafeez) [reply src/app/layout/components/ActionsMenu.tsx:7]: Okay

**Pull request merged**